client/util.js

```diff
@@ -67,7 +67,9 @@ export function ajax(method, url, headers={}, data={}, { fireAndForget = false }
                 }
             };
         }
-
+        if (window.navigator && window.navigator.sendBeacon && fireBeacon) {
+            window.navigator.sendBeacon(url, param(data).replace(/&/g, '%26'));
```

Comment:
Can we avoid pulling in jquery for a fairly lightweight operation?

```
const queryString = Object.keys(data)
    .map(key => `${ encodeURIComponent(key) }=${ encodeURIComponent(data[key]) }`)
    .join('&');
```

Reply:
The suggested solution will fail on nested objects. Also the pulled in module is not jQuery, but a fairly light (900B) module that contains only the param function

Reply:
beaver-logger already expects the data passed to be a flat object when it's a POST. so it would be good to be consistent even if it's a GET.

Reply:
Beaver logger seems to be sending nested data (check slack for example). And sendBeacon is a post. Unfortunately it is currently unable to send data of type application/json

Reply:
Sorry I'm a little confused -- if `sendBeacon` does a post, can we not just do `JSON.stringify(data)`?

We can even pass a content-type header: https://stackoverflow.com/a/41729668

Reply:
That is exactly what I was trying to do and would have saved a lot of headache if it worked on Chrome. Your suggestion (my original approach) works perfectly well on FireFox, in fact, even the original XMLHttpRequest works perfectly for FireFox. However, Chrome has set restrictions on the content types that can be sent using sendBeacon. See this [thread](https://stackoverflow.com/questions/45274021/sendbeacon-api-not-working-temporarily-due-to-security-issue-any-workaround).